tests/phpunit/api/v4/Entity/ConformanceTest.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace api\v4\Entity;
+
+use Civi\Api4\Entity;
+use api\v4\Traits\TableDropperTrait;
+use api\v4\UnitTestCase;
+
+/**
+ * @group headless
+ */
+class ConformanceTest extends UnitTestCase {
+
+  use TableDropperTrait;
+  use \api\v4\Traits\OptionCleanupTrait {
+    setUp as setUpOptionCleanup;
+  }
+
+  /**
+   * @var \api\v4\Service\TestCreationParameterProvider
+   */
+  protected $creationParamProvider;
+
+  /**
+   * Set up baseline for testing
+   */
+  public function setUp() {
+    $tablesToTruncate = [
+      'civicrm_custom_group',
+      'civicrm_custom_field',
+      'civicrm_group',
+      'civicrm_event',
+      'civicrm_participant',
+    ];
+    $this->dropByPrefix('civicrm_value_myfavorite');
+    $this->cleanup(['tablesToTruncate' => $tablesToTruncate]);
+    $this->setUpOptionCleanup();
+    $this->loadDataSet('ConformanceTest');
+    $this->creationParamProvider = \Civi::container()->get('test.param_provider');
+    parent::setUp();
+    // calculateTaxAmount() for contribution triggers a deprecation notice
+    \PHPUnit_Framework_Error_Deprecated::$enabled = FALSE;
```

Comment:
@totten @colemanw for phpunit6 compatibility 
```suggestion
    \PHPUnit\Framework\Error\Deprecated::$enabled = FALSE;
```